src/app/components/application/application-form/application-form.component.ts

```diff
@@ -900,71 +904,95 @@ export class ApplicationFormComponent implements OnInit {
   // Save button
 
   onSaveButtonClicked() {
+    let error = false;
+    if (!this.applicationForm.valid) {
+      error = true;
+      this.utils.showRequiredFieldsError();
+    } else if (!this.validTrees()) {
```

Comment:
`validTrees` approach is a bad name. A better approach could have a logic similar to:

- a list that contains the pairs [validator function `fn(tree)`, error function message `error()`] (we can have multiple checks)
- you iterates the list and executes the validation function `fn(tree)` (e.g. array.find([fn, msg] => fn(tree)))
- if the returned value is defined, the test has found and error, then error is set to true and the function message (e.g. the second component of the returned value) is executed.